GetIntoTeachingApi/Services/CrmService.cs

```diff
@@ -87,20 +87,49 @@ public Candidate MatchCandidate(ExistingCandidateRequest request)
                 return null;
             }
 
-            _service.LoadProperty(entity, new Relationship("dfe_contact_dfe_candidatequalification_ContactId"), context);
-            _service.LoadProperty(entity, new Relationship("dfe_contact_dfe_candidatepastteachingposition_ContactId"), context);
-            _service.LoadProperty(entity, new Relationship("dfe_contact_dfe_servicesubscription_contact"), context);
-            _service.LoadProperty(entity, new Relationship("msevtmgt_contact_msevtmgt_eventregistration_Contact"), context);
+            LoadCandidateRelationships(entity, context);
 
             return new Candidate(entity, this);
         }
 
+        public IEnumerable<Candidate> MatchCandidates(string magicLinkToken)
+        {
+            // Avoids a potentially very expensive query.
+            if (string.IsNullOrEmpty(magicLinkToken))
+            {
+                return new Candidate[0];
+            }
+
+            var context = Context();
+            var entities = _service.CreateQuery("contact", Context())
+                .Where(c => c.GetAttributeValue<string>("dfe_websitemltoken") == magicLinkToken);
+
+            foreach (var entity in entities)
+            {
+                LoadCandidateRelationships(entity, context);
+            }
+
+            return entities.Select(e => new Candidate(e, this));
+        }
+
         public Candidate GetCandidate(Guid id)
         {
-            var entity = _service.CreateQuery("contact", Context())
-                .FirstOrDefault(c => c.GetAttributeValue<EntityReference>("contactid").Id == id);

```

Comment:
will `FirstOrDefault` return `null` when there are no matches? ie is 'default' `null`

Reply:
Yep, there's `First()` which raises when there are no results and `FirstOrDefault()` which returns `null` if there are no matches